### PR TITLE
Replace menu defaults with tokens

### LINKED ISSUE #113962: Replace Menu Defaults with Tokens
The Menu default data is hand-coded based on the spec. We need to change it to use tokens in Material3.

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to replace the menu defaults with tokens. The default icon color in the menu is different from the text color, so a new property `iconColor` is added to `ButtonStyle`.

Some data, such as horizontal and vertical padding come from the specs. The horizontal padding for elements(text, leading/trailing icon) in a row is 12 px, and the vertical padding for the whole menu is 8.

Screen Shot 2022-10-24 at 12 28 03 PM

Fixes: #113962

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/menu_anchor.dart [hunk 1]
```diff
@@ -1021,10 +1023,11 @@ class _MenuItemButtonState extends State<MenuItemButton> {
     // Since we don't want to use the theme style or default style from the
     // TextButton, we merge the styles, merging them in the right order when
     // each type of style exists. Each "*StyleOf" function is only called once.
-    final ButtonStyle mergedStyle =
-        widget.style?.merge(widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context))) ??
-            widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context)) ??
-            widget.defaultStyleOf(context);
+    ButtonStyle mergedStyle = widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context))
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/menu_anchor.dart:1026]: How is this any different from the previous version?  I think it evaluates to the same thing, right?

File: packages/flutter/lib/src/material/menu_anchor.dart [hunk 2]
```diff
@@ -1799,10 +1804,11 @@ class _SubmenuButtonState extends State<SubmenuButton> {
         // TextButton, we merge the styles, merging them in the right order when
         // each type of style exists. Each "*StyleOf" function is only called
         // once.
-        final ButtonStyle mergedStyle =
-            widget.style?.merge(widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context))) ??
-                widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context)) ??
-                widget.defaultStyleOf(context);
+        ButtonStyle mergedStyle = widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context))
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/menu_anchor.dart:1807]: Same here...

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

This looks really good, thanks!

File: packages/flutter/lib/src/material/menu_anchor.dart (hunk 1 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/menu_anchor.dart:1026]: This change is to handle the case when we don't have a custom "themeStyleOf". Previously, if the `widget.style` is not null, but the `widget.themeStyleOf(context)` is null, the first line of code `widget.style?.merge(widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context)))` will just return the `widget.style` without merging with the default theme data. So the button will only have the same style as the default TextButton:) Please let me know if I misunderstood anything.

File: packages/flutter/lib/src/material/menu_anchor.dart (hunk 2 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/menu_anchor.dart:1807]: Similarly here, just to handle the case when we have the `widget.style` but don't have `themeStyleOf`:)

**Review by QuncCccccc — COMMENTED** [review]
@gspencergoog Thanks a lot for your quick review!

File: packages/flutter/lib/src/material/menu_anchor.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/menu_anchor.dart:1026]: Ahh, OK, that makes sense.

**Pull request merged**